src/KeyRegistrar.sol

```diff
@@ -0,0 +1,472 @@
+     * @notice Gets the BN254 G2 public key for an operator with a specific AVS
+     * @param avs Address of the AVS
+     * @param operator Address of the operator
+     * @return pubkeyG2 The BN254 G2 public key
+     */
+    function getBN254KeyG2(address avs, address operator) external view returns (BN254.G2Point memory) {
+        return avsOperatorToBN254KeyG2[avs][operator];
+    }
+
+    /**
+     * @notice Gets the ECDSA public key for an operator with a specific AVS
+     * @param avs Address of the AVS
+     * @param operator Address of the operator
+     * @return pubkey The ECDSA public key
+     */
+    function getECDSAKey(address avs, address operator) external view returns (bytes memory) {
+        return avsOperatorToECDSAKey[avs][operator];
+    }
+
+    /**
+     * @notice Gets the operator address from a key hash for a specific AVS
+     * @param avs Address of the AVS
+     * @param curveType Type of curve
+     * @param keyHash Hash of the key
+     * @return operator The operator address
+     */
+    function getOperatorFromKeyHash(
+        address avs,
+        CurveType curveType,
+        bytes32 keyHash
+    ) external view returns (address) {
+        return avsKeyHashToOperator[avs][curveType][keyHash];
+    }
+
+    /**
+     * @notice Gets the key hash for an operator with a specific AVS
+     * @param avs Address of the AVS
+     * @param operator Address of the operator
+     * @param curveType Type of curve
+     * @return keyHash The key hash
+     */
+    function getKeyHash(
+        address avs,
+        address operator,
+        CurveType curveType
+    ) external view returns (bytes32) {
+        return avsOperatorToKeyHash[avs][operator][curveType];
+    }
+
+    /**
+     * @notice Registers an ECDSA key for an operator with a specific AVS
+     * @param avs Address of the AVS
+     * @param operator Address of the operator
+     * @param pubkey ECDSA public key bytes
+     */
+    function _registerECDSAKey(
+        address avs,
+        address operator,
+        bytes memory pubkey
+    ) internal {
+        if (pubkey.length != 65) {
+            revert InvalidKeyFormat();
+        }
+
+        bytes32 keyHash = keccak256(pubkey);
+        address existingOperator = avsKeyHashToOperator[avs][CurveType.ECDSA][keyHash];
+        
+        if (existingOperator != address(0) && existingOperator != operator) {
+            revert KeyAlreadyRegistered();
+        }
+
+        avsOperatorToECDSAKey[avs][operator] = pubkey;
+        avsOperatorToKeyHash[avs][operator][CurveType.ECDSA] = keyHash;
+        avsKeyHashToOperator[avs][CurveType.ECDSA][keyHash] = operator;
+    }
+
+    /**
+     * @notice Registers a BN254 key for an operator with a specific AVS
+     * @param avs Address of the AVS
+     * @param operator Address of the operator
+     * @param pubkey BN254.G1Point and BN254.G2Point encoded as bytes
+     * @return The registered BN254.G1Point
+     */
+    function _registerBN254Key(
```

Comment:
they need to provide a signature here as well to prevent https://xn--2-umb.com/22/bls-signatures/#rogue-key-attack